Spring boot application with file operations

vikasreddy-elixr opened the pull request.

vikasreddy-elixr committed:
added constants class

vikasreddy-elixr committed:
Springboot application implementing the API of uploading the file and get the file by UUID and username apis.

vikasreddy-elixr committed:
Added a new line in the ErrorResponse class.

premsrj commented on build.gradle:
```diff
@@ -21,10 +15,14 @@ repositories {
 dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-data-mongodb'
     implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'com.jamierf.db-table:db-table-core:0.1'
```
Could you explain where this is used?

vikasreddy-elixr replied:
This isn't used.
Removed this dependency.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/exceptionhandler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,63 @@
+    private ErrorResponse buildErrorResponse(String message) {
+        return ErrorResponse.builder().success(Constants.FAILURE).message(message).build();
+    }
+}
```
Missing new line at the end of the file

vikasreddy-elixr replied:
added new line.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/controller/FileController.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @PostMapping(value = "/upload")
+    public ResponseEntity<?> FileUpload(@RequestParam("file") MultipartFile multipartfile, @RequestParam("username") @NotEmpty String userName, FileInfo fileInfo) {
```
Why do you need this fileInfo parameter?

vikasreddy-elixr replied:
fileInfo is required in service class, removed the fileInfo and initialized in Service class.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/model/FileInfo.java:
```diff
@@ -0,0 +1,24 @@
+
+    @MongoId
+    @GeneratedValue
+    private UUID id= UUID.randomUUID();
```
Code style should be proper

```suggestion
    private UUID id = UUID.randomUUID();
```

vikasreddy-elixr replied:
reformatted

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/reporsitory/GlobalRepository.java:
```diff
@@ -0,0 +1,15 @@
+
+
+@Repository
+public interface GlobalRepository extends MongoRepository<FileInfo, String> {
```
Why is it called GlobalRepository when it is dealing with file information?

vikasreddy-elixr replied:
I have named it as general.
I'll change that to FileRepository.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/responses/SuccessResponse.java:
```diff
@@ -0,0 +1,43 @@
+
+@Data
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class SuccessResponse {
```
Since you seem to have three different constructors with different set of parameters, don't you think it would be better to use a Builder pattern?

vikasreddy-elixr replied:
Done with it.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,96 @@
+
+    @Override
+    public ResponseEntity<?> getFileById(String id) {
+        if ((!id.matches("[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"))){
```
Wouldn't it be easier to try to convert the id to UUID and if it fails, you can then throw the exception?

vikasreddy-elixr replied:
Done with that.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,96 @@
+        }
+    }
+    @Override
+    public ResponseEntity<?> fileUpload(MultipartFile multipartFile, FileInfo fileInfo, String userName) {
```
I believe I mentioned in the demo that you should use the id you get from the database as the filename to save it?

vikasreddy-elixr replied:
saved the file with id  as fileName

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,96 @@
+        if ((!id.matches("[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"))){
+            return new ResponseEntity<>(new ErrorResponse(Constants.FAILURE,Constants.ERROR_UUID_FORMAT),HttpStatus.BAD_REQUEST);
+        }
+            FileInfo getFile = globalRepository.findById(id).orElseThrow(() -> new NotFoundException(Constants.ERROR_NOT_FOUND));
```
Code style is very important.
```suggestion
        FileInfo getFile = globalRepository.findById(id).orElseThrow(() -> new NotFoundException(Constants.ERROR_NOT_FOUND));
```

vikasreddy-elixr replied:
code Reformatted.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,96 @@
+                String data = new String(Files.readAllBytes(Paths.get(filePath + fileName)));
+                return new ResponseEntity<>(new SuccessResponse(Constants.SUCCESS, getFile.getUserName(), getFile.getLocalDateTime(), fileName, data), HttpStatus.OK);
+            } catch (IOException e) {
+                return new ResponseEntity<>(new ErrorResponse(Constants.FAILURE, e.getMessage()), HttpStatus.BAD_REQUEST);
```
If it is an IOException from the server, why is it returning a bad request? Doesn't bad request indicate that the user made an error while calling the API?

vikasreddy-elixr replied:
changed Bad request to Not Found.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,96 @@
+            return new ResponseEntity<>(new ErrorResponse(Constants.FAILURE,Constants.ERROR_NO_DATA_FOUND),HttpStatus.NOT_FOUND);
+        }
+
+        return new ResponseEntity<>(new SuccessResponse(Constants.SUCCESS,userName,globalRepository
```
Right now you are making two different reads on the database. Database operations have an overhead. Wouldn't it be better to call globalRepository.findByUserName(userName) and save the value to a variable, then use this variable in the if condition and the response entity creation?

vikasreddy-elixr replied:
Done with that prem

vikasreddy-elixr committed:
used builder pattern instead of constructor in the Success Response class.
removed unused dependencies in the build.gradle.
Renamed globalRepository as FileRepository.

vikasreddy-elixr committed:
removed the dead code.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/responses/SuccessResponse.java:
```diff
@@ -0,0 +1,33 @@
+    private String fileName;
+    private String content;
+    private List<FileInfoDTO> files;
+    private SuccessResponse file;
```
Why do you have SuccessResponse inside SuccessResponse? It would mean that object has another SuccessResponse inside it, which would have another and so on..
If you are only going to keep the file details here, make a separate class for it.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/responses/SuccessResponse.java:
```diff
@@ -0,0 +1,33 @@
+    private List<FileInfoDTO> files;
+    private SuccessResponse file;
+
+    public SuccessResponse(String status, SuccessResponse build) {
```
You are using a builder pattern, remove this constructor

vikasreddy-elixr replied:
Removed  the Constructor.
it was used to Alter the response of the api.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,95 @@
+                String data = new String(Files.readAllBytes(Paths.get(filePath + uuid)));
+                return new ResponseEntity<>(new SuccessResponse(Constants.SUCCESS, SuccessResponse.builder().userName(getFile.getUserName()).uploadTime(getFile.getLocalDateTime()).fileName(fileName).content(data).build()), HttpStatus.OK);
+            } catch (IOException e) {
+                return new ResponseEntity<>(new ErrorResponse(Constants.FAILURE, e.getMessage()), HttpStatus.NOT_FOUND);
```
This is an exception. not a not_found. You should be returning an internal server error

vikasreddy-elixr replied:
Changed with it.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,95 @@
+
+    @Override
+    public ResponseEntity<?> getFileById(UUID uuid) {
+        FileInfo getFile = fileRepository.findById(String.valueOf(uuid)).orElseThrow(() -> new NotFoundException(Constants.ERROR_NOT_FOUND));
```
I mean accept the id as a string itself to this method. Inside this method try to convert it to uuid and if that fails return a bad request

vikasreddy-elixr replied:
That is done in Controller itself.

premsrj reviewed (changes requested):
Add Sreejith also to review

vikasreddy-elixr committed:
removed the dead code.

kssreejith1994 commented on build.gradle:
```diff
@@ -21,10 +15,13 @@ repositories {
 dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-data-mongodb'
     implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'org.eclipse.persistence:javax.persistence:2.2.1'
```
Could you please explain the use of this dependency? Mongodb starter dependency includes the required annotations for db operations right?

vikasreddy-elixr replied:
persistence is used because we have used persistence.generatedValue

kssreejith1994 commented on build.gradle:
```diff
@@ -21,10 +15,13 @@ repositories {
     implementation 'org.springframework.boot:spring-boot-starter-data-mongodb'
     implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'org.eclipse.persistence:javax.persistence:2.2.1'
+    implementation 'junit:junit:4.13.2'
```
Junit is part of of the starter test dependency. So is this really needed?

vikasreddy-elixr replied:
Junit is not required.
I have removed it.

vikasreddy-elixr replied:
I think since we are working on the Junit part we need it.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/controller/FileController.java:
```diff
@@ -0,0 +1,46 @@
+
+    @PostMapping(value = "/upload")
+    public ResponseEntity<?> FileUpload(@RequestParam("file") MultipartFile multipartfile, @RequestParam("username") @NotEmpty String userName) {
+        if (Objects.equals(multipartfile.getContentType(), "text/plain")) {
```
What does this line do? If it is of any other format what happens?

vikasreddy-elixr replied:
i throws an comment of File format not supported.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/controller/FileController.java:
```diff
@@ -0,0 +1,46 @@
+        return new ResponseEntity<>(new ErrorResponse(Constants.FAILURE, Constants.ERROR_FILE_FORMAT_NOT_SUPPORTED), HttpStatus.BAD_REQUEST);
+    }
+
+    @GetMapping("/file/{id}")
```
Maintain consistency. If you are using value attribute as in line 28, use that everywhere. Else remove from everywhere. That way code would be same across

vikasreddy-elixr replied:
Removed the value attribute.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/controller/FileController.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @GetMapping("/file/user/{userName}")
+    public ResponseEntity<?> getFilesByUserName(@PathVariable(name = "userName") String userName) {
```
you dont need to mention name here in the @pathvariable. It is required only when the variable that holds the name i different from the path variable

vikasreddy-elixr replied:
removed the name.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/exceptionhandler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,65 @@
+
+    @ExceptionHandler(ConstraintViolationException.class)
+    public ResponseEntity<?> ConstraintViolationException(ConstraintViolationException exception) {
+        return new ResponseEntity<>(buildErrorResponse(exception.getMessage()), HttpStatus.BAD_REQUEST);
```
Constraint violations can have multiple errors. How is that handled?

premsrj replied:
Provide a reply to this comment

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/controller/FileController.java:
```diff
@@ -0,0 +1,46 @@
+import java.util.Objects;
+import java.util.UUID;
+
+@Validated
```
What is this annotation used for?

vikasreddy-elixr replied:
it is used to validate the parameters.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/exceptionhandler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    @ExceptionHandler(NotFoundException.class)
+    public ResponseEntity<?> NotFoundException() {
```
Dont we need the thrown exception to be caught here?

premsrj replied:
Provide a reply to this comment

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/model/FileInfo.java:
```diff
@@ -0,0 +1,35 @@
+    private String userName;
+    @NotEmpty
+    private String fileName;
+    private LocalDateTime localDateTime = LocalDateTime.now();
```
This should be a final variable

vikasreddy-elixr replied:
Done with the change.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/model/FileInfo.java:
```diff
@@ -0,0 +1,35 @@
+    public FileInfo(String id ,String userName, String fileName) {
+        this.userName = userName;
+        this.fileName = fileName;
+        this.id = UUID.fromString(id);
```
What if the passed string is not an UUID? Constructor should not be used for such operations. You should validate it before calling this and pass the UUID directly.

vikasreddy-elixr replied:
solved.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/model/FileInfoDTO.java:
```diff
@@ -0,0 +1,16 @@
+import java.util.UUID;
+
+@Getter
+@Setter
```
Replace with @Data annotation instead of getter and setter

vikasreddy-elixr replied:
Replaced.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/reporsitory/FileRepository.java:
```diff
@@ -0,0 +1,15 @@
+
+
+@Repository
+public interface FileRepository extends MongoRepository<FileInfo, String> {
```
The id is an UUID right? So why String here?

vikasreddy-elixr replied:
Replaced with UUID.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/responses/ErrorResponse.java:
```diff
@@ -0,0 +1,14 @@
+
+    private String success;
+    private String message;
+}
```
New line here

vikasreddy-elixr replied:
new line added.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/responses/ResponseForGetById.java:
```diff
@@ -0,0 +1,15 @@
+import lombok.Data;
+
+@Data
+public class ResponseForGetById{
```
Cant we add the all args constructor annotation here?

vikasreddy-elixr replied:
yeah.
Added @AllArgsConstructor

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/model/FileInfoDTO.java:
```diff
@@ -0,0 +1,16 @@
+
+@Getter
+@Setter
+public class FileInfoDTO {
```
Cant we use the @Builder annotation and build it rather than using the constructor

vikasreddy-elixr replied:
Done with it.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,99 @@
+    final String filePath;
+    {
+        try {
+            filePath = new ClassPathResource("/static/  ").getFile().getAbsolutePath();
```
Can you explain what is this line doing? Why is there a separate block added here?

vikasreddy-elixr replied:
That is used to get the path of the Project folder.
and that is throwing an exception and is handled using try catch.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,99 @@
+    public ResponseEntity<?> fileUpload(MultipartFile multipartFile, String userName) {
+
+        FileInfo fileInfo;
+        try {
```
Try should not cover unnecessary lines. Only the line that is expected to throw the exception should be inside the try block. Identify that and use try only on that line/lines

vikasreddy-elixr replied:
written the lines that doesn't throw exceptions outside the try block.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,99 @@
+            fileInfo.setFileName(multipartFile.getOriginalFilename());
+            fileInfo.setUserName(userName);
+            fileRepository.save(fileInfo);
+            Files.copy(multipartFile.getInputStream(), Paths.get(filePath + fileInfo.getId()), StandardCopyOption.REPLACE_EXISTING);
```
What if file write failed due to some permission issue (try writing the file to any system folder in your laptop) ? Will the saved file info still be present in the db?

vikasreddy-elixr replied:
It throws Not found exception.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,99 @@
+        UUID uuid = UUID.fromString(id);
+        FileInfo getFile = fileRepository.findById(String.valueOf(uuid)).orElseThrow(() -> new NotFoundException(Constants.ERROR_NOT_FOUND));
+        String fileName = getFile.getFileName();
+        File file = new File(filePath + uuid);
```
Use a StringBuilder instead of using the + operator to join strings

vikasreddy-elixr replied:
done with the Change

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,99 @@
+        if (file.exists()) {
+            try {
+                String data = new String(Files.readAllBytes(Paths.get(filePath + uuid)));
+                return new ResponseEntity<>(new ResponseForGetById(Constants.SUCCESS, Response.builder()
```
Create a separate builder method to build the response.

vikasreddy-elixr replied:
Created the builder method.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,99 @@
+        if (fileInfoList.isEmpty()) {
+            return new ResponseEntity<>(new ErrorResponse(Constants.FAILURE, Constants.ERROR_NO_DATA_FOUND), HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(Response.builder().status(Constants.SUCCESS).userName(userName).files((fileInfoList.stream().map(this::convertDataIntoDTO).collect(Collectors.toList()))).build(), HttpStatus.OK);
```
What if someone deletes the file from the upload location and then try to get the file details? File details will be there but actual file will not be there right? So should the availability of the file should be returned to the user?

vikasreddy-elixr replied:
It throws FileNotFound Exception.

vikasreddy-elixr committed:
Testing Part has been done and the changes done in the response and the exception handling.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/controller/FileController.java:
```diff
@@ -0,0 +1,47 @@
+
+@RestController
+public class FileController {
+    FileService fileService;
```
Make this final

vikasreddy-elixr replied:
Done with the change.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -0,0 +1,103 @@
+
+@Service
+public class FileServiceImpl implements FileService {
+    FileRepository fileRepository;
```
Make this final

vikasreddy-elixr replied:
Done with the change.

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/controller/FileControllerTest.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    @Test
+    public void test_GetFileById() throws Exception {
```
Need to add another test method for 404 of Get File by Id too

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/controller/FileControllerTest.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    @Test
+    public void test_GetFileByUsername() throws Exception {
```
Need to add another test method for 404 of Get File by Username too

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/services/FileServiceImplTest.java:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    @Test
+    public void getFileById() {
```
Again need to test 404 flow in another test method

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/services/FileServiceImplTest.java:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    @Test
+    public void getFileByUsername() {
```
Need to test 404 flow in another test method

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/controller/FileControllerTest.java:
```diff
@@ -0,0 +1,106 @@
+                        .contentType(MediaType.TEXT_PLAIN)
+                        .param("username", "userName"))
+                .andExpect(MockMvcResultMatchers.status().isOk()).andReturn();
+        Assertions.assertEquals(200, result.getResponse().getStatus());
```
Is this line needed when you are doing
.andExpect(MockMvcResultMatchers.status().isOk())
?

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/controller/FileControllerTest.java:
```diff
@@ -0,0 +1,106 @@
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andReturn();
+        Assertions.assertEquals(200, result.getResponse().getStatus());
```
Is this line needed when you are doing
.andExpect(MockMvcResultMatchers.status().isOk())

vikasreddy-elixr replied:
removed the assertEquals.

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/controller/FileControllerTest.java:
```diff
@@ -0,0 +1,106 @@
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(json.toString()))
+                .andExpect(MockMvcResultMatchers.status().isOk()).andReturn();
+        Assertions.assertEquals(200, result.getResponse().getStatus());
```
Is this line needed when you are doing
.andExpect(MockMvcResultMatchers.status().isOk())

vikasreddy-elixr replied:
removed the assertEquals.

kssreejith1994 commented on Dockerfile:
```diff
@@ -0,0 +1,4 @@
+FROM openjdk:17
+LABEL maintainer="vikas"
+ADD spring-boot-application-with-docker.jar spring-boot-docker-application.jar
+ENTRYPOINT ["java", "-jar", "spring-boot-docker-application.jar"]
```
New line here

vikasreddy-elixr replied:
Added new Line.

kssreejith1994 commented on src/main/java/com/example/springapplicationforfileoperation/services/FileServiceImpl.java:
```diff
@@ -33,67 +33,71 @@ public FileServiceImpl(FileRepository fileRepository) {
     {
         try {
-            filePath = new ClassPathResource("/static/  ").getFile().getAbsolutePath();
+            filePath = new ClassPathResource("/static/ ").getFile().getAbsolutePath();
```
Do we need to really do this? Can you explain why this block is separated out?

vikasreddy-elixr replied:
Resolved it.

vikasreddy-elixr committed:
testing for 404 has been done and made fileByUserName to return the existence of the file.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/contants/Constants.java:
```diff
@@ -0,0 +1,13 @@
+    public static final String FILE_PATH = "${C:\\Users\\hp\\IdeaProjects\\Springapplicationforfileoperation\\build\\resources\\main\\static\\}";
+    public static final boolean TRUE = true;
+    public static final boolean FALSE = false;
+}
```
Missing new line

vikasreddy-elixr replied:
new line added.

premsrj commented on src/main/java/com/example/springapplicationforfileoperation/contants/Constants.java:
```diff
@@ -0,0 +1,13 @@
+    public static final String ERROR_NO_DATA_FOUND = "Data with Given User Name Not Found";
+    public static final String ERROR_FILE_FORMAT_NOT_SUPPORTED = "File Format Not Supported or Input File is Missing";
+    public static final String ERROR_UNEXPECTED_TYPE = "Unexpected internal server error";
+    public static final String FILE_PATH = "${C:\\Users\\hp\\IdeaProjects\\Springapplicationforfileoperation\\build\\resources\\main\\static\\}";
```
Will this path work from within a docker container?

vikasreddy-elixr replied:
altered this
file to be saved in the project folder itself.

premsrj commented on src/main/resources/application.properties:
```diff
@@ -1 +1,5 @@
-
+spring.data.mongodb.uri=mongodb://localhost:27017/trainingPoc
```
Can you access mongodb through localhost when running in a docker container?

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/SpringApplicationForFileOperationApplicationTests.java:
```diff
@@ -0,0 +1,19 @@
+
+
+@RunWith(MockitoJUnitRunner.class)
+@SpringBootTest
```
You can remove this test class, not used

vikasreddy-elixr replied:
removed it.

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/controller/FileControllerTest.java:
```diff
@@ -0,0 +1,124 @@
+
+        Mockito.when(fileController.getFileById(id))
+                .thenReturn(new ResponseEntity<>(HttpStatus.NOT_FOUND));
+        ResponseEntity<?> result = fileController.getFileById(id);
```
Test not valid, you are directly calling the method. You should use mockMvc to perform the test similar to a network call

vikasreddy-elixr replied:
Done with this.

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/controller/FileControllerTest.java:
```diff
@@ -0,0 +1,124 @@
+        String userName = "userName";
+        when(fileController.getFilesByUserName(userName))
+                .thenReturn(new ResponseEntity<>(HttpStatus.NOT_FOUND));
+        ResponseEntity<?> result = fileService.getFilesByUserName(userName);
```
Again, test not valid, you are calling the method directly.

vikasreddy-elixr replied:
Done with this.

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/services/FileServiceImplTest.java:
```diff
@@ -0,0 +1,122 @@
+        String id = "a8580190-a275-44f6-b691-12a7902ae15e";
+        when(fileRepository.findById(UUID.fromString(id))).thenThrow(new NotFoundException(Constants.ERROR_NOT_FOUND));
+        try{
+           fileService.getFileById(id);
```
The test should fail if fileService.getFileById(id); does not throw a NotFoundException. You need to use an Assert.fail here and force the test to fail if no exception is thrown

vikasreddy-elixr replied:
Done with this.

premsrj commented on src/test/java/com/example/springapplicationforfileoperation/services/FileServiceImplTest.java:
```diff
@@ -0,0 +1,122 @@
+        String userName = "userName";
+        when(fileRepository.findByUserName(userName)).thenThrow(new NotFoundException(Constants.ERROR_NOT_FOUND));
+        try {
+            fileService.getFilesByUserName(userName);
```
The test should fail if fileService.getFileById(id); does not throw a NotFoundException. You need to use an Assert.fail here and force the test to fail if no exception is thrown

vikasreddy-elixr replied:
Done with this.

vikasreddy-elixr committed:
Dockerfile and Docker Compose files has been created.
